atox/src/main/kotlin/ActionReceiver.kt

```diff
@@ -73,6 +74,11 @@ class ActionReceiver : BroadcastReceiver() {
             Log.e(TAG, "Got intent without required key $KEY_CONTACT_PK $intent")
             return
         }
+        if (runBlocking { !contactExists(pk.string()) }) {
```

Comment:
Room (by default) blocks database queries on the main thread as that isn't great for application behaviour/performance. This cheats a bit by blocking the main thread while running the query on another thread, which means it has the same effect.

Is this really needed after dismissing the notifications on contact deletion? If so, is there a way of doing this without blocking the main thread on db queries?

Reply:
It's needed for preventing a possible crash. All we need is to check if the contact exists before we precede with handling the action. I will look into doing it the other way around: executing all of the code of the receiver in another thread and somehow run the main thread specific stuff like dismissing notifications in the main thread.

Reply:
I understand that getting here after deleting a contact would crash, but is it feasible that that would happen even after dismissing the notifications on contact delete?

If this can't be done in a clean way I guess this can do for now, even if it's sad that we'll block the main thread on every intent broadcast. Rewriting all handling of intent broadcasts for an unlikely edge case doesn't feel good.

Reply:
I see in https://github.com/evilcorpltd/aTox/blob/master/atox/src/main/kotlin/ActionReceiver.kt#L95:L98 that we can dismiss the notifications from another thread. Maybe all we need is:
```kotlin
override fun onReceive(context: Context, intent: Intent) = scope.launch {
// .....
}
```

The `acceptCall` function already does that and uses the main thread to display toasts so why not moving it up a bit in the hierarchy? It would also get rid of those multiple `scope.launch` calls.

Reply:
Yeah, that should work. Will just need some testing. :P iirc there aren't any guarantees about background jobs created in BroadcastReceivers running (I read this after the acceptCall stuff), but I don't think that's broken, and we only run this while the main application is already running, so it should be fine.

Reply:
Didn't know about that. Hopefully it'll be OK.